### PR TITLE
Updating en-GB site nav

### PR DESCRIPTION
**Pull request by jamesshedden**
**DNM**:

- [x] Add final 'Solutions' dropdown `nav_title` values to `messages/en`, currently using placeholders:

image

---

### New nav contents & layout

This PR updates the en-GB site nav with a new structure — 'Solutions', 'Who uses us?', 'Help & resources' and 'About us' dropdowns — and re-arranges some of the existing links to fit in with this new structure. We also have a new nav layout for the site with main content on the left, and secondary content (eg 'About us', 'More') sitting on the right:

image

### Developer & API docs messages refactoring

I've also refactored the way we refer to some of the external developer resources in the messages files (and then consequently throughout the templates) as there was some inconsistency with naming & structure that was causing a bit of confusion.

### `header.js` refactoring

Lastly I've taken the opportunity to refactor the file that contains the nav, `header.js` to improve readability & maintainability — when I started working on this template it was initially quite hard to parse due to number of inline functions (for generating popovers, using `classnames` etc) which I refactored into their own functions outside of the template.

### CODE REVIEW

**Comment by jamesshedden** [comment]
@harrison Hey, be awesome to get a pair of eyes on this — unfortunately I won't be around to follow up with the feedback as am off on hols, maybe one @megaslug can help us get over the line once you've taken a look? 😊

File: app/components/header/header.js [hunk 1]
```diff
@@ -27,207 +34,435 @@ class Header extends React.Component {
...
+      <Popover className='popover--large'
+      toggle={(
+        <a className={ this.getLinkClassName() }>
+          <div className={ this.getPopoverLinkClassName() }>
+            <Message pointer='header.our_products' />
+          </div>
+        </a>
+      )}>
+        <ul className='u-text-xxs u-padding-Vxs'>
+          { _.map(PRODUCT_DROPDOWN_LINKS, (link, index) => this.renderProductLink(link, index)) }
```
Comment (harrison) [inline app/components/header/header.js]: would inline `PRODUCT_DROPDOWN_LINKS` and do something like
```
['pro', 'features', 'partner_with_us'].map(this.renderProductLink)
```

File: app/components/header/header.js [hunk 2]
```diff
@@ -27,207 +34,435 @@ class Header extends React.Component {
     availableLocales: PropTypes.array.isRequired,
   };
 
-  render() {
-    const { currentLocale, availableLocales } = this.context;
+  renderProductLink(link) {
```
Comment (harrison) [inline app/components/header/header.js]: maybe inline as a function in `renderProductPopover`? think it makes sense to keep related pieces of code close to each other, makes it clear it's not meant to be reused as is.

File: app/components/header/header.js [hunk 3]
```diff
@@ -27,207 +34,435 @@ class Header extends React.Component {
...
-        <IfLinkExists to='partner_with_us' tagName='li'>
-          <hr className='u-margin-Vxs' />
-
-          <Link to='partner_with_us' className='site-header__product-link u-link-clean'>
-            <span className='site-header__product-link-title u-text-semi'><Message pointer='partner_with_us.nav_title' /></span>
-            <p className='u-color-dark-gray'><Message pointer='partner_with_us.explainer' /></p>
-          </Link>
-        </IfLinkExists>
-      </ul>
+  getLoginButtonClassName() {
```
Comment (harrison) [inline app/components/header/header.js]: not entirely sure about these class names used only once, is it that bad assigning them at the top of the render function? :) `getLinkClassName` and `getPopoverLinkClassName` seem legit.

**Comment by megaslug** [comment]
Hey @harrison, I've addressed your comments, could you have another look? Thanks

**Comment by harrison** [comment]
@megaslug nice work! Looks good to me 👍 on shipping after fixing the conflicts. Also squash and merge so we get less noise on dev :)

Happy to pair on cleaning up the branch or has James already walked you through our normal rebase process?
